src/inmanta/ast/type.py

```diff
@@ -161,6 +200,43 @@ def __eq__(self, other: object) -> bool:
             return NotImplemented
         return self.element_type == other.element_type
 
+    def is_primitive(self) -> bool:
+        return self.element_type.is_primitive()
+
+    def corresponds_to(self, pytype: type[object]) -> bool:
+        if not typing_inspect.is_optional_type(pytype):
+            return False
+        # remove options from union
+        other_types = [tp for tp in typing_inspect.get_args(pytype) if not typing_inspect.is_optional_type(tp)]
+        # Typing union conveniently returns the base type if that is the only element
+        return self.element_type.corresponds_to(typing.Union[*other_types])
+
+    def as_python_type_string(self) -> "str | None":
+        return f"{self.element_type.as_python_type_string()} | None"
+
+    def to_python(self, instance: object) -> "object":
+        if isinstance(instance, NoneValue):
+            return None
+        return self.element_type.to_python(instance)
+
+    def has_custom_to_python(self) -> bool:
+        return self.element_type.has_custom_to_python()
+
+
+class Any(Type):
```

Comment:
If we keep this, I'd like a docstring that clearly states that while this type represents the any type, similar to Python's, the `Any` class itself is neither the top nor the bottom type in the type hierarchy.

Turns out it's the same for `typing.Any`, but still I think it's good to point out.

Reply:
I now realized the same actually goes for e.g. typedefs / unions / ..., so it was a bit naive of me to think that this might be the case in the first place. Docstring doesn't hurt, but I no longer think it required.